### PR TITLE
Break circular imports

### LINKED ISSUE #195: RF: dandi/cli to avoid circular import
probably to move all helper functions in a dedicated `cli/base.py` etc.

Otherwise this demonstrates them:
```
$> python -m pytest -v --doctest-modules dandi/cli
======================================================================================================= test session starts =======================================================================================================
platform linux -- Python 3.8.5, pytest-4.6.9, py-1.8.1, pluggy-0.13.0 -- /home/yoh/proj/dandi/dandi-cli/venvs/dev3/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/yoh/proj/dandi/dandi-cli/.hypothesis/examples')
rootdir: /home/yoh/proj/dandi/dandi-cli, inifile: pytest.ini
plugins: mock-3.2.0, cov-2.10.0, xdist-1.32.0, timeout-1.3.3, forked-1.1.3, hypothesis-4.36.2
collected 8 items / 2 errors / 6 selected                                                                                                                                                                                         

============================================================================================================= ERRORS ==============================================================================================================
___________________________________________________________________________________________ ERROR collecting dandi/cli/cmd_download.py ____________________________________________________________________________________________
dandi/cli/cmd_download.py:4: in <module>
    from .command import devel_option, main, map_to_click_exceptions
dandi/cli/command.py:222: in <module>
    from .cmd_download import download  # noqa: E402, F401
E   ImportError: cannot import name 'download' from partially initialized module 'dandi.cli.cmd_download' (most likely due to a circular import) (/home/yoh/proj/dandi/dandi-cli/dandi/cli/cmd_download.py)
___________________________________________________________________________________________ ERROR collecting dandi/cli/cmd_register.py ____________________________________________________________________________________________
dandi/cli/cmd_register.py:3: in <module>
    from .command import (
dandi/cli/command.py:224: in <module>
    from .cmd_register import register  # noqa: E402, F401
E   ImportError: cannot import name 'register' from partially initialized module 'dandi.cli.cmd_register' (most likely due to a circular import) (/home/yoh/proj/dandi/dandi-cli/dandi/cli/cmd_register.py)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
===================================================================================================== 2 error in 0.16 seconds =============================================================
```

### PR DESCRIPTION
**Pull request by jwodder**
Closes #195

### CODE REVIEW

File: dandi/cli/cmd_download.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import os
 
 import click
-from .command import devel_option, main, map_to_click_exceptions
+from .base import devel_option, map_to_click_exceptions
 
 
-@main.command()
+@click.command()
```
Comment (yarikoptic) [inline dandi/cli/cmd_download.py:7]: shouldn't it be a part of main to take advantage of all of its setup etc?

Wouldn't we just move everything BUT following
```
#
# Commands in the main group
#
from .cmd_ls import ls  # noqa: E402, F401
from .cmd_organize import organize  # noqa: E402, F401
from .cmd_upload import upload  # noqa: E402, F401
from .cmd_download import download  # noqa: E402, F401
from .cmd_validate import validate  # noqa: E402, F401
from .cmd_register import register  # noqa: E402, F401
```
from `command.py` into `base.py` and adjust the imports to use `from .base`?  alternative could be just to move those lines into e.g. `cli/ui.py` and import that one upstairs where command is imported now?

Comment (jwodder) [reply dandi/cli/cmd_download.py:7]: Take advantage of what setup?  Using `@main.command` vs. `@click.command` + `main.add_command()` should make no difference.  Deciding between how I organized the command imports versus your suggestions seems like just a stylistic choice.

Comment (yarikoptic) [reply dandi/cli/cmd_download.py:7]: I have missed `main.add_command()` invocations... ok, let's proceed with your stylistic choice.

**Comment by yarikoptic** [comment]
merging #192 before this bigger RF caused conflicts :-/

**Comment by jwodder** [comment]
@yarikoptic Conflicts resolved.

**Comment by yarikoptic** [comment]
Great, thanks! CI isn't happy though, didn't look inside yet

**Comment by jwodder** [comment]
I think that's because of #199.  I believe I fixed the upload errors, but another failing test has come up as well.

**Pull request merged**